teamscale_client/client.py

```diff
@@ -127,61 +124,30 @@ def add_findings_group(self, name, mapping_pattern):
         Returns:
             requests.Response: request's response
         """
-        url = self.get_global_service_url('external-findings-group')
-        finding_group = {'groupName': name, 'mapping': mapping_pattern}
-        payload = [finding_group]
-        if self.supports_at_least_api_version_5:
-            url = "%s/%s" % (url, name)
-            payload = finding_group
-
-        return self.put(url, payload)
+        url = "%s/%s" % (self.get_global_service_url('external-findings-group'), name)
+        return self.put(url, {'groupName': name, 'mapping': mapping_pattern})
 
     def add_finding_descriptions(self, descriptions):
         """Adds descriptions of findings.
 
         Args:
             descriptions (list): List of :class:`FindingDescription` to add to Teamscale.
         Returns:
-            requests.Response: request's response
+            str: 'success' if all descriptions were added.
         """
-        @responses.activate
-        def success_response():
-            """ Returns a mock success response for clients when using Teamscale server
-                with minimum supported api version 5
-            """
-            success_url = 'http://success_url'
-            responses.add(responses.GET, success_url, status=200, body='success')
-            return requests.get(success_url)
-
-        payload = []
+        base_url = self.get_global_service_url('external-findings-description')
         for finding_description in descriptions:
-            some_description = {}
+            some_description = dict()
             some_description['typeId'] = finding_description.typeid
             some_description['description'] = finding_description.description
             some_description['enablement'] = finding_description.enablement
-            if self.supports_at_least_api_version_5:
-                some_description['name'] = finding_description.name
-                response = self._add_finding_description(some_description)
-                result = "Description (typeId -> %s). Upload result: %s" % (finding_description.typeid, response.text)
-                print (result)
-            else:
-                payload.append(some_description)
-
-        if self.supports_at_least_api_version_5:
-            return success_response()
-
-        return self.put(self.get_global_service_url('add-external-finding-descriptions'), payload)
+            some_description['name'] = finding_description.name
+            url = "%s/%s" % (base_url, finding_description.typeid)
+            response = self.put(url, some_description)
```

Comment:
Why can't we upload them all at once anymore?

Reply:
Because currently the ElementListService exposes only putElement(String elementName, T element). For bulk, we shall need to extend this method.

Reply:
Ok, please create a ticket for this